server/src/main/java/org/elasticsearch/common/settings/SecureSetting.java

```diff
@@ -169,6 +201,12 @@ SecureString getFallback(Settings settings) {
             if (fallback != null) {
                 return fallback.get(settings);
             }
+            if (DiscoveryNode.isStateless(settings)) {
```

Comment:
With this change any secure setting will fallback to an unsecure setting when stateless is enabled. 

I wonder if we should have another static `SecureSetting.secureStringFallback()` method that would instantiate a `SecureSetting` instance with a `FallbackStringSetting` fallback instance, something like:

```java
    public static Setting<SecureString> statelessSecureString(String name, Property... properties) {
        return new SecureStringSetting(name, new FallbackStringSetting(name, properties), properties);
    }
```

so that settings that fallback to unsecure in the context of stateless are clearly identified?

Reply:
@tlrx , with the latest comment on Ryan's suggestion, I can attempt a SecureSettings interface implementation just for Stateless that reads all secure settings from the .yml (rather than just specific ones). That goes in contrast to your suggestion here to have only specific secure settings fall back to .yml. Please tell me if this would still be a blocker concern for you if we attempt the suggested SecureSettings implementation.